### PR TITLE
tests: new variable used to configure the kernel command line in nested tests

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The change is to use an env var to configure the command line in the pc-gadget

This is an example:
NESTED_EXTRA_CMDLINE='dangerous rd.systemd.debug-shell=1' spread -debug google-nested:ubuntu-22.04-64:tests/nested/manual/uc20-fde-hooks

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Thank you, we should probably include this in some README or HACKING that this can be useful

**Review by valentindavid — COMMENTED** [review]
I am not totally sure why `console=ttyS0` is added when `NESTED_EXTRA_CMDLINE` is defined.

File: tests/lib/nested.sh [hunk 1]
```diff
@@ -619,10 +619,19 @@ defaults:
...
-                echo 'console=ttyS0 snapd.debug=1 systemd.journald.forward_to_console=1 ' > pc-gadget/cmdline.extra
+                GADGET_EXTRA_CMDLINE="snapd.debug=1 systemd.journald.forward_to_console=1"
+            fi
+            if [ -n "$NESTED_EXTRA_CMDLINE" ]; then
+                GADGET_EXTRA_CMDLINE="$GADGET_EXTRA_CMDLINE $NESTED_EXTRA_CMDLINE"
+            fi
+
+            if [ -n "$GADGET_EXTRA_CMDLINE" ]; then
+                echo "Configuring command line parameters in the gadget snap: \"console=ttyS0 $GADGET_EXTRA_CMDLINE\""
+                echo "console=ttyS0 $GADGET_EXTRA_CMDLINE" > pc-gadget/cmdline.extra
```
Comment (Meulengracht) [inline tests/lib/nested.sh:634]: Valentin already pointed this out, but it seems to me the logic here has changed, the `console=ttyS0` was moved  from the DEBUG_TO_SERIAL to here where it's now only included if GADGET_EXTRA_CMDLINE is not set, was this intentional?

Comment (sergiocazzolato) [reply tests/lib/nested.sh:634]: right, I'll update that, thanks for the review

**Review by Meulengracht — APPROVED** [review]
Thanks for a really cool addition! +1

File: tests/lib/nested.sh [hunk 2]
```diff
@@ -619,10 +619,19 @@ defaults:
...
                 # visibility what happens when a machine fails to boot
-                echo 'console=ttyS0 snapd.debug=1 systemd.journald.forward_to_console=1 ' > pc-gadget/cmdline.extra
+                GADGET_EXTRA_CMDLINE="console=ttyS0 snapd.debug=1 systemd.journald.forward_to_console=1"
+            fi
+            if [ -n "$NESTED_EXTRA_CMDLINE" ]; then
+                GADGET_EXTRA_CMDLINE="$GADGET_EXTRA_CMDLINE $NESTED_EXTRA_CMDLINE"
+            fi
+
+            if [ -n "$GADGET_EXTRA_CMDLINE" ]; then
+                echo "Configuring command line parameters in the gadget snap: \"console=ttyS0 $GADGET_EXTRA_CMDLINE\""
```
Comment (mardy) [inline tests/lib/nested.sh:633]: I guess you meant
```suggestion
                echo "Configuring command line parameters in the gadget snap: \"$GADGET_EXTRA_CMDLINE\""
```

**Pull request merged**